src/mono/mono/metadata/debug-mono-ppdb.c

```diff
@@ -456,57 +469,90 @@ mono_ppdb_is_embedded (MonoPPDBFile *ppdb)
 	return ppdb->is_embedded;
 }
 
-static int 
-mono_ppdb_get_seq_points_internal (const char* ptr, MonoSymSeqPoint **seq_points, int *n_seq_points, int docidx, MonoImage *image, MonoPPDBFile *ppdb, GPtrArray **sfiles, char **source_file, int **source_files, GPtrArray **sindexes, gboolean read_doc_value)
+static int
+mono_ppdb_get_seq_points_internal (MonoImage *image, MonoPPDBFile *ppdb, MonoMethod* method, int method_idx, char **source_file, GPtrArray **source_file_list, int **source_files, MonoSymSeqPoint **seq_points, int *n_seq_points)
 {
+	MonoTableInfo *tables = image->tables;
+	guint32 cols [MONO_METHODBODY_SIZE];
+	const char *ptr;
+	const char *end;
+	MonoDebugSourceInfo *docinfo;
+	int i, size, docidx, iloffset, delta_il, delta_lines, delta_cols, start_line, start_col, adv_line, adv_col;
+	gboolean first = TRUE, first_non_hidden = TRUE;
 	GArray *sps;
 	MonoSymSeqPoint sp;
-	int iloffset = 0;
-	int start_line = 0;
-	int start_col = 0;
-	int delta_cols = 0;
-	gboolean first_non_hidden = TRUE;
-	int adv_line, adv_col;
-	int size = mono_metadata_decode_blob_size (ptr, &ptr);
-	const char* end = ptr + size;
-	MonoDebugSourceInfo *docinfo;
-	gboolean first = TRUE;
+	GPtrArray *sfiles = NULL;
+	GPtrArray *sindexes = NULL;
+	if (source_file)
+		*source_file = NULL;
+	if (source_file_list)
+		*source_file_list = NULL;
+	if (source_files)
+		*source_files = NULL;
+	if (seq_points)
+		*seq_points = NULL;
+	if (n_seq_points)
+		*n_seq_points = 0;
+
+	if (source_file_list)
+		*source_file_list = sfiles = g_ptr_array_new ();
+	if (source_files)
+		sindexes = g_ptr_array_new ();
+
+	if (!method->token || table_info_get_rows (&tables [MONO_TABLE_METHODBODY]) == 0)
+		return -1;
+
+	MonoTableInfo *methodbody_table = &tables [MONO_TABLE_METHODBODY];
+	if (G_UNLIKELY (method_idx - 1 >= table_info_get_rows (methodbody_table))) {
+		char *method_name = mono_method_full_name (method, FALSE);
+		g_error ("Method idx %d is greater than number of rows (%d) in PPDB MethodDebugInformation table, for method %s in '%s'. Likely a malformed PDB file.",
```

Comment:
I think right now this is ok since we only have hot reload method body updates - we can't add new methods.  But in the future this probably won't work right.

I think this check should be moved up to the (non-EnC) caller, or dropped entirely (or dropped if we have any updates).